src/TriggerActivityMakerChannelAdjacency.cpp

```diff
@@ -0,0 +1,241 @@
+/**
+ * @file TriggerActivityMakerChannelAdjacency.cpp
+ *
+ * This is part of the DUNE DAQ Application Framework, copyright 2021.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+
+#include "triggeralgs/ChannelAdjacency/TriggerActivityMakerChannelAdjacency.hpp"
+#include "TRACE/trace.h"
+#define TRACE_NAME "TriggerActivityMakerChannelAdjacencyPlugin"
+#include <vector>
+#include <math.h>
+
+using namespace triggeralgs;
+
+void
+TriggerActivityMakerChannelAdjacency::operator()(const TriggerPrimitive& input_tp,
+					     std::vector<TriggerActivity>& output_ta)
+{
+
+  // Add useful info about recived TPs here for FW and SW TPG guys.
+  if (m_print_tp_info){
+    TLOG(1) << " ########## m_current_window is reset ##########\n"
+	    << " TP Start Time: " << input_tp.time_start << ", TP ADC Sum: " << input_tp.adc_integral
+	    << ", TP TOT: " << input_tp.time_over_threshold << ", TP ADC Peak: " << input_tp.adc_peak
+	    << ", TP Offline Channel ID: " << input_tp.channel << "\n";
+  }
+  
+  // 0) FIRST TP =====================================================================
+  // The first time operator() is called, reset the window object.
+  if (m_current_window.is_empty()) {
+    m_current_window.reset(input_tp);
+    m_primitive_count++;
+    return;
+  }
+  
+  // If the difference between the current TP's start time and the start of the window
+  // is less than the specified window size, add the TP to the window.
+  bool adj_pass = 0; // sets to true when adjacency logic is satisfied
+  bool window_filled = 1; // sets to true when window is ready to test the adjacency logic
+  if ((input_tp.time_start - m_current_window.time_start) < m_window_length) {
+    m_current_window.add(input_tp);
+    window_filled = 0;
+    TLOG(1) << "m_current_window.time_start " << m_current_window.time_start << "\n";
+  }
+  
+  else {
+    TPWindow win_adj_max;
+
+    bool ta_found = 1;
+    while (ta_found) {
+      
+      // make m_current_window_tmp a copy of m_current_window and clear m_current_window
+      TPWindow m_current_window_tmp = m_current_window;
+      m_current_window.clear();
+      
+      // make m_current_window a new window of non-overlapping tps (of m_current_window_tmp and win_adj_max)                                                                                 
+      for (auto tp : m_current_window_tmp.inputs) {
+	bool new_tp = 1;
+	for (auto tp_sel : win_adj_max.inputs) {
+	  if (tp.channel == tp_sel.channel) { new_tp = 0; break; }
+	}
```

Comment:
This piece will exclude any TPs that occur on the same set of channels at a later time in the same window, such as half of the fork in the example below. 
![event-271](https://github.com/DUNE-DAQ/triggeralgs/assets/5084895/18f450e4-49b2-47f2-9b92-5a2cee2983e5)
A worse case would be if two separate tracks appear on the same channels at the same time window.
![event-898](https://github.com/DUNE-DAQ/triggeralgs/assets/5084895/200a75b5-884f-4831-9531-d320588c606d)

I admit that this will require much more thought than is necessary for this first PR, but it is a limitation to be aware of.

Reply:
yes, I agree.